tests/analyse.py

```diff
@@ -62,3 +64,13 @@ def analyze_mining(deployment, last_layer, layers_per_epoch, layer_avg_size, tot
         blocks_created_per_layer = blocks_in_relevant_layers / (last_layer - layers_per_epoch)
         wanted_avg_block_per_node = max(1, int(layer_avg_size / total_pods))
         assert blocks_created_per_layer / wanted_avg_block_per_node == 1
+
+
+def analyze_propagation(deployment, last_layer, max_block_propagation, max_atx_propagation):
+    for i in range(last_layer):
+        propagation, msg_time = queries.layer_block_max_propagation(deployment, last_layer)
+        print(propagation, msg_time)
+        assert msg_time < datetime.timedelta(seconds=max_block_propagation)
```

Comment:
why not return the propagation and do the asserts in the test ?

Reply:
analyse is the file containing or mesh assertions,as you can see from other tests